plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.ts

```diff
@@ -158,6 +162,13 @@ export class TaskWorker {
 
   stop() {
     this.stopWorkers = true;
+    if (this.options?.gracefulShutdown) {
```

Comment:
Is `stop` externally visible public api? If not, should it not be blocking (made async), and the caller block on it too?

Reply:
Yep, it's visible 😟

Reply:
@benjdlamben how would you feel about making it async in a minor release, saying "please await if you care about when tasks finish"? This really should not break anybody right 🤔

Reply:
yeah, I mean it's non-breaking to do it that way, it really should be connected to the `lifecycleService` anyways I feel.

Reply:
@freben you mean there's no need for the config value but instead make it the default behaviour for the `stop` function?

Reply:
No I was only referring to making the method async and blocking (and the caller awaiting that), nothing else.

Spawning a little free floating piece of work won't have any effect in practise, I think.

Reply:
I changed it now to async. Also mention in the changeset and changed it to minor version update.